test/e2e/replica_set.go

```diff
@@ -249,3 +267,87 @@ func rsConditionCheck(f *framework.Framework) {
 	}
 	Expect(err).NotTo(HaveOccurred())
 }
+
+func testAdoptMatchingOrphans(f *framework.Framework) {
+	name := "pod-adoption"
+	By(fmt.Sprintf("Given a Pod with a 'name' label %s is created", name))
+	p := f.PodClient().CreateSync(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  name,
+					Image: nginxImageName,
+				},
+			},
+		},
+	})
+
+	By(fmt.Sprintf("When a replicaset with a matching selector is created"))
+	replicas := int32(2)
+	rsSt := newRSWithSelector(name, replicas, map[string]string{"name": name}, name, nginxImageName, name)
+	rs, err := f.ClientSet.Extensions().ReplicaSets(f.Namespace.Name).Create(rsSt)
+	Expect(err).NotTo(HaveOccurred())
+	// Cleanup the ReplicaSet when we are done.
+	defer func() {
+		if err := framework.DeleteReplicaSet(f.ClientSet, f.InternalClientset, f.Namespace.Name, rs.Name); err != nil {
+			framework.Logf("Failed to cleanup ReplicaSet %v: %v.", rs.Name, err)
+		}
+	}()
+
+	By("Then the orphan pod is adopted")
+	// List the replicaset pods, making sure we observe all the replicas.
+	pods, err := framework.PodsCreated(f.ClientSet, f.Namespace.Name, rs.Name, replicas)
```

Comment:
Indeed, sorry, something was wrong with the way I was doing the call and it was failing. 

However, following your reasoning about the filter by label vs checking the controller owner in the release case, I think that listing the pods that match the label won't help us in this case either, the pod to be adopted will always match the criteria. I've modified the test to check for the owner in this case too, please have another look.

Thanks!